Number_types/include/CGAL/Quotient.h

```diff
@@ -685,12 +785,239 @@ template < class NT > class Real_embeddable_traits_quotient_base< Quotient<NT> >
 
     class To_interval
       : public CGAL::cpp98::unary_function< Type, std::pair< double, double > > {
+
       public:
+
+        bool are_correct_bounds( const double l, const double u, const Type& x ) const {
+
+          if (
+            CGAL::abs(l) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(u) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(l) == 0.0 ||
+            CGAL::abs(u) == 0.0) {
+            return true;
+          }
+          CGAL_assertion(CGAL::abs(l) != std::numeric_limits<double>::infinity());
+          CGAL_assertion(CGAL::abs(u) != std::numeric_limits<double>::infinity());
+          CGAL_assertion(CGAL::abs(l) != 0.0);
+          CGAL_assertion(CGAL::abs(u) != 0.0);
+
+          const Type lb(l), ub(u);
+          CGAL_assertion(lb <= ub);
+          CGAL_assertion(lb <= x);
+          CGAL_assertion(ub >= x);
+          return lb <= ub && lb <= x && ub >= x;
+        }
+
+        std::pair< Interval_nt<>, int64_t > get_interval_exp( NT& x ) const {
+
+          CGAL_assertion(x >= 0);
+          int64_t d = 0;
+          double l = 0.0, u = 0.0;
+          const int64_t n = static_cast<int64_t>(msb(x)) + 1;
+          const int64_t num_dbl_digits = std::numeric_limits<double>::digits;
+
+          if (n > num_dbl_digits) {
+            d = n - num_dbl_digits;
+            x = x >> d;
+            const NT y = x + 1;
```

Comment:
The +1 is still done in type NT, not in type `int64_t` or similar, and there are 2 casts from NT to double. I don't know if it would help, but since you wrote "I think using what you proposed is better. I have fixed that."...

Reply:
I did that in the other implementation but did not in this one. Sorry. I think now it should be ok.